pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -94,6 +94,13 @@ type CephClusterList struct {
 }
 
 // ClusterSpec represents the specification of Ceph Cluster
+// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.mon)))))",message="external mode enabled cluster cannot have mon field in CR"
```

Comment:
I'd suggest similar wording in all the messages. Just talk about the settings instead of a "mode" in the CR
```suggestion
// +kubebuilder:validation:XValidation:rule="!has(self.external) || (has(self.external) && ((!has(self.external.enable) || self.external.enable == false )|| (self.external.enable == true && (!has(self.mon)))))",message="external clusters cannot define the mon settings"
```